datacube/index/postgres/tables/_core.py

```diff
@@ -154,6 +167,9 @@ def update_schema(engine):
     if not engine.execute("SELECT 1 FROM pg_type WHERE typname = 'float8range'").scalar():
         engine.execute(TYPES_INIT_SQL)
 
+    if not _pg_column_exists(engine, schema_qualified('dataset_location'), 'archived'):
```

Comment:
We should add `_LOG.info` lines like some of the other changes here, to log that it's making/made a schema change.